lib/evmone/instructions.hpp

```diff
@@ -154,87 +159,132 @@ inline constexpr auto invalid = stop_impl<EVMC_INVALID_INSTRUCTION>;
 
 inline void add(StackTop stack) noexcept
 {
-    stack.top() += stack.pop();
+    stack[1].val = stack[0].val + stack[1].val;
```

Comment:
We are potentially leaking the memory for the stack item at index 0, at least until that stack location is used for something else.

Reply:
i find the stack pop/puhing in this code somewhat tricky...was the original code not leaky? if so, why?

Reply:
The original code did not dynamically allocate memory for stack items.

Reply:
because now it's a struct containing the uint256, plus a shared_ptr, which is dynamically allocated elsewhere right?

Reply:
Yes, the shared_ptr dynamically allocates memory, which is not de-allocated when shared_ptr is on the stack.